### PR TITLE
update(SMARTGRIDDASHBOARD parser) - fix fetch_production

### PR DESCRIPTION
**Pull request by mathilde-daugy**
## Issue

There is an issue with the `fetch_production` function in the parser. Investigation shows that there is an exchange between GB-NIR and IE that is unaccounted for meaning that there are some difference between `total generation` (from the dashboard) and the calculated `total generation` (demand - net imports)

**IE**
image
**GB-NIR**
image

See issue [ELE-3449](https://linear.app/electricitymaps/issue/ELE-3449/gb-nir)

## Description

- Update of `fetch_production`: `unknown` is now `total generation`- `wind` 

### Preview
**IE**
image

**GB-NIR**
image

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

___

## **Type**
Enhancement

___

## **Description**
- The `fetch_production` function in the SMARTGRIDDASHBOARD parser has been updated to correct an issue with the calculation of `unknown` production. 
- Previously, `unknown` was calculated as `demand` - `wind` - `exchange`. However, this calculation was found to be incorrect.
- The new calculation for `unknown` is `total generation` - `wind`, which aligns with the actual data.
- This change should resolve the discrepancy between the `total generation` from the dashboard and the calculated `total generation`.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>SMARTGRIDDASHBOARD.py</strong><dd><code>Update SMARTGRIDDASHBOARD Parser to Correct Production Calculation</code></dd></summary>
<hr>

parsers/SMARTGRIDDASHBOARD.py
- Replaced `demand_data` with `total_generation` in `fetch_production` <br>  function.
- Removed `exchange_data` from `fetch_production` function.
- Updated the calculation of `unknown` in the `productionMix` to be <br>  `total generation` - `wind`.
- Removed unused imports and breakpoints.
    </details>
  </td>
  <td><a href="https://github.com/electricitymaps/electricitymaps-contrib/pull/6417/files#diff-202eb9e172e952cdcbc60124abaf696d294cac1a1122b2de9a1495a618b428ce">+14/-25</a>&nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```

<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        

</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        

</details></td></tr>

<tr><td><details> <summary><strong> Inline File Walkthrough 💎</strong></summary><hr>

For enhanced user experience, the `describe` tool can add file summaries directly to the "Files changed" tab in the PR page.
This will enable you to quickly understand the changes in each file, while reviewing the code changes (diffs).

To enable inline file summary, set `pr_description.inline_file_summary` in the configuration file, possible values are:
- `'table'`: File changes walkthrough table will be displayed on the top of the "Files changed" tab, in addition to the "Conversation" tab.
- `true`: A collapsable file comment with changes title and a changes summary for each file in the PR.
- `false` (default): File changes walkthrough will be added only to the "Conversation" tab.
<tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `describe` tool can be configured with extra instructions, to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Notice that the general structure of the description is fixed, and cannot be changed. Extra instructions can change the content or style of each sub-section of the PR description.

Examples for extra instructions:
```
[pr_description] 
extra_instructions="""
- The PR title should be in the format: '<PR type>: <title>'
- The title should be short and concise (up to 10 words)
- ...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.

</details></td></tr>

<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 

</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.

</details>

### CODE REVIEW

File: parsers/SMARTGRIDDASHBOARD.py
Comment (VIKTORVAV99) [inline parsers/SMARTGRIDDASHBOARD.py:155]:
```suggestion
        wind_prod = float(wind_event_dt[0]["Value"]) if wind_event_dt[0]["Value"] else 0
```

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
The current implementation has a none check issue that needs to be resolved, I provided a suggestion for it.

**Comment by VIKTORVAV99** [comment]
@florianscheidl have you explored estimating exchanges like this where we have all the other needed data to do so?

**Comment by mathilde-daugy** [comment]
I've had a quick look and it does look like we could estimate the GB-NIR-IE from the available data but it would require more in depth investigation

**Comment by VIKTORVAV99** [comment]
> I've had a quick look and it does look like we could estimate the GB-NIR-IE from the available data but it would require more in depth investigation. In the meantime, this change enables us to get total generation for the island which would already be a win. :)

I kind of already did that in a slightly hacky way with totalGenerationForcasts ;) (apparently we don't have a totalGeneration parser type)

**Review by VIKTORVAV99 — APPROVED** [review]
Approving since it works now but I'll let you decide if we should merge it or not before florain or robin has taken a look.

**Comment by florianscheidl** [comment]
Hey @VIKTORVAV99.
We looked into this a few weeks ago (see these [slides](https://docs.google.com/presentation/d/1NGpE_sciYkMDkNqb-FR97ZOciMPskyYMpKzOlg6fnes/edit#slide=id.g158aa0f95e7_0_0)). 
We have contacted people from EirGrid and are waiting for a response. In the meantime, I agree with Mathilde that it makes sense to implement this change and further investigate exchange estimation once we have heard back 🙂

**Review by florianscheidl — APPROVED** [review]
LGTM, thanks 👍

**Comment by VIKTORVAV99** [comment]
I was thinking of doing something more direct like `exchange = all IE generation + IE imports/exports - all GB-NIR generation + all GB-NIR imports/exports` that should be the missing flow between IE and GB-NIR right? (assuming no losses)

**Comment by mathilde-daugy** [comment]
> I was thinking of doing something more direct like `exchange = all IE generation + IE imports/exports - all GB-NIR generation + all GB-NIR imports/exports` that should be the missing flow between IE and GB-NIR right? (assuming no losses)

Yes i have that in mind as well but i would like to do a bit more research because the difference to make sure that this approximation is consistent across the whole time series. I will most likely have the time to look at it tomorrow or thursday

**Pull request merged**
